### DISCUSSION TITLE
Structs

### DISCUSSION THREAD

**Discussion by brendorsilva** [balta-io/2801#34] [Perguntas e Respostas] [answered]
Olá, estou acompanhando o curso de C# fundamentos para aprimorar minha base, já trabalho com c# em projetos asp.net a algum tempo e nunca utilizei structs, pelo que percebe, se parece muito com o class.

Gostaria de saber tem alguma diferença além de um ser value types e o outro ser reference types, e quando eu devo escolher entre um e outro ?

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @brendorsilva, como vai?

Você chegou a ver a aula sobre structs?
https://balta.io/player/assistir/5ef614da-366d-6156-d049-694800000000/bde7b21f-b979-4982-bac9-000028010542

Esta e a próxima aula eu falo sobre isto!

**Reply by brendorsilva** [reply]
Eu assisti as duas aulas sobre o structs, fiquei na dúvida por que em meus projetos eu uso as "class" para difinir minhas propriedades e meus metódos, ai gostaria de saber se tem mais alguma diferença além do type de cada uma.

**Reply by andrebaltieri** [reply]
Bom dia, @brendorsilva, como vai?

Tem sim, a memória volátil tem 3 áreas, estática, heap e stack. O stack, como comento no curso, armazena os dados de forma mais organizada, porém é mais limitada em relação ao heap.

Desta forma, objetos maiores normalmente vão para o heap, como são o caso das classes no C#. Quando criamos uma classe, podemos usar todo poder da OOP como herança. Enquanto nos structs ficamos limitados a estruturas de dados.

Embora os structs estejam evoluindo também ainda há uma limitação física (E explicações mais complexas para isto). Nos processadores 32bits é 1mb, nos 64bits é 4mb. Enquanto o heap pode usar quanto tiver na máquina :)

Vou tentar escrever algo sobre isto, espero ter ajudado 💜
